go/vt/vttablet/tabletconn/tablet_conn.go

```diff
@@ -36,9 +38,31 @@ var (
 
 var (
 	// TabletProtocol is exported for unit tests
-	TabletProtocol = flag.String("tablet_protocol", "grpc", "how to talk to the vttablets")
+	//
+	// (TODO|andrew): rewrite tests so this does not need to be exported.
+	TabletProtocol = "grpc"
```

Comment:
While debugging the fix in `healthcheck_test`, I found that the following change will let us un-export `TabletProtocol`.

- Pros: let's us protect the global variable, and probably reduce the chance of accidentally writing racy tests.
- Cons: more verbose

```diff
--- a/go/vt/vttablet/tabletconn/tablet_conn.go
+++ b/go/vt/vttablet/tabletconn/tablet_conn.go
- var TabletProtocol = "grpc"
+ var tabletProtocol = "grpc"

- func registerFlags(fs *pflag.FlagSet) {
-    fs.StringVar(&TabletProtocol, "tablet_protocol", "...")
+   fs.StringVar(&tabletProtocol, "tablet_protocol", "...")
}

--- a/go/vt/discovery/healthcheck_test.go
+++ b/go/vt/discovery/healthcheck_test.go
+const testName = "go.vt.discovery.test"
+
 func init() {
        tabletconn.RegisterDialer("fake_gateway", tabletDialer)
 
+       servenv.OnParseFor(testName, tabletconn.RegisterFlags)
+       tmp := os.Args[:]
+       os.Args = []string{testName}
+       servenv.ParseFlags(testName)
+       os.Args = tmp
+
        // log error
-       if err := flag.Set("tablet_protocol", "fake_gateway"); err != nil {
+       if err := pflag.Set("tablet_protocol", "fake_gateway"); err != nil {
                log.Errorf("failed to set flag \"tablet_protocol\" to \"fake_gateway\":%v", err)
        }
```

thoughts? I can do this everywhere if we prefer this over the "export the variable" approach we currently use (cc @deepthi )

Reply:
I vote yes. More test code is better than exporting a variable that we don't really want to export.

Reply:
https://github.com/vitessio/vitess/issues/11009